flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/BufferWritingResultPartition.java

```diff
@@ -330,8 +330,16 @@ private BufferBuilder appendBroadcastDataForRecordContinuation(
     private void createBroadcastBufferConsumers(BufferBuilder buffer, int partialRecordBytes)
             throws IOException {
         try (final BufferConsumer consumer = buffer.createBufferConsumerFromBeginning()) {
+            int desirableBufferSize = Integer.MAX_VALUE;
             for (ResultSubpartition subpartition : subpartitions) {
-                subpartition.add(consumer.copy(), partialRecordBytes);
+                int subPartitionBufferSize = subpartition.add(consumer.copy(), partialRecordBytes);
+                desirableBufferSize =
+                        subPartitionBufferSize > 0
+                                ? Math.min(desirableBufferSize, subPartitionBufferSize)
+                                : desirableBufferSize;
+            }
+            if (desirableBufferSize != Integer.MAX_VALUE) {
```

Comment:
Just a minor comment: it is up to you but this condition is not really needed here because buffer.trim(MAX_VALUE) is not a problem it just set the buffer size to the maximum possible value which is ok. But it is not a mistake if you left this condition here.